File: tests/machine/x/dart/user_type_literal.dart
class Person {
  String name;
  int age;
  Person(this.name, this.age);
}

class Book {
  String title;
  Person author;
  Book(this.title, this.author);
}

var book = Book('Go', Person('Bob', 42));

void main() {
  print(book.author.name);
}

File: tests/machine/x/lua/user_type_literal.lua
Person = {}
Person.__index = Person
function Person.new(o)
    o = o or {}
    setmetatable(o, Person)
    return o
end

Book = {}
Book.__index = Book
function Book.new(o)
    o = o or {}
    setmetatable(o, Book)
    return o
end

book = {title="Go", author={name="Bob", age=42}}
print(book.author.name)
